base/parse.jl

```diff
@@ -167,6 +167,29 @@ float(x::AbstractString) = parse(Float64,x)
 
 float{S<:AbstractString}(a::AbstractArray{S}) = map!(float, similar(a,typeof(float(0))), a)
 
+function parse{T<:AbstractFloat}(::Type{T}, s::AbstractString, base::Integer)
+    res = zero(T)
+    2 <= base <= 9 || throw(ArgumentError("invalid base: base must be 2 ≤ base ≤ 9, got $base"))
+    b = Float64(base)
+    n, Exponent = 0,0
+    pointfound = false
+    for i in eachindex(s)
+        if s[i] == '.' 
+            Exponent = i-1
+            pointfound = true
+            continue
+        end
+        n += 1
+        d = parse(Int,s[i])
```

Comment:
You can probably use `parse(Int, s[i], base)` here. This will work for even more bases and you don't need to check `d < base` yourself.